squidDrone/src/com/com_message_buffer.cpp

```diff
@@ -0,0 +1,25 @@
+#include "com_message_buffer.hpp"
+#include "com_types.hpp"
+
+namespace com {
+  auto ComMessageBuffer::PutData(std::array<std::uint8_t, 32> &data) noexcept -> types::ComError{
+    if(data_.size() > max_queue_len){
+      return types::ComError::COM_BUFFER_OVERFLOW;
+    }
+    data_.push(data);
+    return types::ComError::COM_OK;
+  }
+
+  auto ComMessageBuffer::GetData() noexcept -> std::array<std::uint8_t, 32>{
+    std::array<std::uint8_t, 32> queue_item = {0};
+    if(!data_.empty()){
```

Comment:
you have BufferIsEmpty. Use it, otherwise this is duplicate code

Reply:
Is this not more efficient, than using a member function?

Reply:
The conflict readabillity(code duplication, name shortening) vs performance is always won by readabillity unless it is absolutely necessary to optimize.

Rule of thumb: Premature optimization is the root of all evil - Donald Knuth

Reply:
Fixed with 7ad53cc798ba88f5254fe13be7e3c41cf32165f6